test/tests/bootstrap/test_redfish10_os_bootstrap_parallel.py

```diff
@@ -0,0 +1,196 @@
+rackhdconfig = fit_common.rackhdapi('/api/2.0/config')['json']
+rackhdhost = "http://" + str(rackhdconfig['apiServerAddress']) + ":" + str(rackhdconfig['apiServerPort'])
+
+
+# this routine polls a task ID for completion
+def wait_for_task_complete(taskid, retries=60):
+    for dummy in range(0, retries):
+        result = fit_common.rackhdapi(taskid)
+        if result['status'] != 200:
+            log.error(" " + result['text'])
+            return False
+        if result['json']['TaskState'] == 'Running' or result['json']['TaskState'] == 'Pending':
+            if fit_common.VERBOSITY >= 2:
+                print "OS Install workflow state: {}".format(result['json']['TaskState'])
+            fit_common.time.sleep(30)
+        elif result['json']['TaskState'] == 'Completed':
+            if fit_common.VERBOSITY >= 2:
+                print "OS Install workflow state: {}".format(result['json']['TaskState'])
+            log.info_5(" " + result['text'])
+            return True
+        else:
+            log.error(" " + result['text'])
+            return False
+    log.error(" Workflow Timeout: " + result['text'])
+    return False
+
+# helper routine for selecting OS image path by matching proxy path
+def proxySelect(tag):
+    for entry in rackhdconfig['httpProxies']:
+        if tag in entry['localPath']:
+            return entry['localPath']
+    return ''
+
+# helper routine to return the task ID associated with the running bootstrap workflow
+def node_taskid(osname, version):
+    for entry in NODE_STATUS:
+        if NODE_STATUS[entry]['os'] == osname and NODE_STATUS[entry]['version'] == version:
+            return NODE_STATUS[entry]['id']
+    return ""
+
+# run all bootstraps if regression group is specified or no group is specified
+if "regression" in sys.argv or "-a" not in sys.argv:
+    oslist = [
+        {"os":"ESXi", "version":"5.5", "path":"/ESXi/5.5"},
+        {"os":"ESXi", "version":"6.0", "path":"/ESXi/6.0"},
+        {"os":"CentOS", "version":"6.5", "path":"/CentOS/6.5"},
+        {"os":"CentOS+KVM", "version":"6.5", "path":"/CentOS/6.5"},
+        {"os":"CentOS", "version":"7", "path":"/CentOS/7.0"},
+        {"os":"RHEL+KVM", "version":"7", "path":"/RHEL/7.0"},
+        {"os":"RHEL", "version":"7", "path":"/RHEL/7.0"},
+        {"os":"CentOS+KVM", "version":"7", "path":"/CentOS/7.0"}
+    ]
+    nodeindex = 0
+    for item in oslist:
+        # if OS proxy entry exists in RackHD config, run bootstrap against selected node
+        if proxySelect(item['path']) and nodeindex < len(NODECATALOG):
+            #delete active workflows for specified node
+            fit_common.cancel_active_workflows(NODECATALOG[nodeindex])
+            payload_data = {
+                            "osName": item['os'],
+                            "version": item['version'],
+                            "repo": rackhdhost + proxySelect(item['path']),
+                            "rootPassword": "1234567",
```

Comment:
ESXi password policy is more strict than other OS, see https://communities.vmware.com/docs/DOC-31941, so this rootPassword actually violates the policy, and it will eventually lead to the rootPassword doesn't been set correctly. Of course, this will not fail the workflow so you will not aware of such problem.

To fix this potential problem, I recommend you change the rootPassword to a complex one.